apiserver/watcher.go

```diff
@@ -180,16 +180,13 @@ func (aw *SrvAllWatcher) translateApplication(info multiwatcher.EntityInfo) para
 		return nil
 	}
 
-	// Get the application status from the cache if it is unset.
+	// If the application status is unset, then set it to unknown. It is
+	// expected that downstream clients (model cache, pylibjuju, jslibjuju)
+	// correctly interpret the unknown status from the unit status. If the unit
+	// status is not found, then fall back to unknown.
+	// If a charm author has set the application status, then show that instead.
 	applicationStatus := multiwatcher.StatusInfo{Current: status.Unknown}
-	if orig.Status.Current == status.Unset {
```

Comment:
But if we fix the all-watcher backing store update, won't this be correct and not even require the implication-from-units logic in the Python client?